### PR TITLE
[klima data] Displays token details details page context information

### LINKED ISSUE #1735: [klima data] Issue|Retire button state selection not being maintained
This page: https://carbon-git-staging-klimadao.vercel.app/off-chain-vs-on-chain

Issues:
- If I select the toggle button Retired option, and then click a Details -> page, when I go back to the main page, the Retired selection is not maintain, it switches back to Issued

### LINKED ISSUE #1762: [klima data] Token Details - toggle selections need to be indicated on Detail pages
This page: https://carbon-git-staging-klimadao.vercel.app/token-details?tab=toucan
Note: applies to all tabs

Issues:
- When I make filter / toggle button selections, the charts on the main page update
Screenshot 2023-10-25 at 3 09 20 PM

- If I open a chart Details --> the correct chart is shown, i.e. the filter / toggle button selections are respected. However, I have no indication anymore what the selections are
- Need to indicate the toggle button selections on Details --> pages

### PR DESCRIPTION
**Pull request by biwano**
## Description

- Makes option widgets save state as an URL query param
- Display option widgets state in token details details pages

## Related Ticket

Resolves #1735 
Resolves #1762  

## Changes

| Before  | After  |
|---------|--------|
|image |image|

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

**Review by jabby09 — CHANGES_REQUESTED** [review]
On token details page, the passing of parameters is not working consistently.

https://carbon-git-carbon-qa1735-klimadao.vercel.app/details/state-of-digital-carbon/moss?pool=all&status=bridged&since=last7d
Screenshot 2023-10-30 at 4 26 33 PM

Also example
Screenshot 2023-10-30 at 4 28 28 PM
Screenshot 2023-10-30 at 4 28 55 PM

It also looks like the URL parameters and concatenating in some strange way - here is a mix of moss and toucan since I visited those pages:
https://carbon-git-carbon-qa1735-klimadao.vercel.app/token-details?tab=toucan&option_moss_0=last7d&option_toucan_0=nct&option_toucan_1=last30d

**Comment by biwano** [comment]
> On token details page, the passing of parameters is not working consistently.
> 
> https://carbon-git-carbon-qa1735-klimadao.vercel.app/details/state-of-digital-carbon/moss?pool=all&status=bridged&since=last7d Screenshot 2023-10-30 at 4 26 33 PM
> 
> Also example Screenshot 2023-10-30 at 4 28 28 PM Screenshot 2023-10-30 at 4 28 55 PM
> 
Hi @jabby09,

The cards State of xxx digital carbon and Breakdown of xxx pooled do not take the parameters into account (same behaviour than in the legacy app). Maybe it is weird from the user perspective and we should move the parameters below those charts. @theprofessora ?

> It also looks like the URL parameters and concatenating in some strange way - here is a mix of moss and toucan since I visited those pages: https://carbon-git-carbon-qa1735-klimadao.vercel.app/token-details?tab=toucan&option_moss_0=last7d&option_toucan_0=nct&option_toucan_1=last30d

Yes because it is the same page and we keep the state for the different tabs. So when you switch from Toucan to C3 then back from C3 to Toucan, the Toucan state is the one you have seen before. But we can change this behaviour if necessary.

File: carbon/components/OptionsSwitcher/index.tsx [hunk 1]
```diff
@@ -1,37 +1,78 @@
...
+  // The widget initial value is the first option value or the value from the url query
+  let initialValue = props.value || props.options[0].value;
+  if ("undefined" !== typeof window) {
+    const url = new URL(window.location.href);
+    initialValue = (url.searchParams.get(props.name) as T) || initialValue;
+  }
   const [selected, setSelected] = useState<T>(
     props.value || props.options[0].value
   );
+  const readonly = props.readonly === undefined ? false : props.readonly;
```
Comment (sprrwhwk) [inline carbon/components/OptionsSwitcher/index.tsx:31]: There's an [operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) for that! 😄 
```suggestion
  const readonly = props.readonly ?? false;
```

File: carbon/components/OptionsSwitcher/index.tsx [hunk 2]
```diff
@@ -1,37 +1,78 @@
...
+  // Notify the parent of the initial value
+  useEffect(() => {
+    const url = new URL(window.location.href);
+    initialValue = (url.searchParams.get(props.name) as T) || initialValue;
+    setSelected(initialValue);
+    console.log(initialValue);
+    if (!readonly && props.onSelectionChange) {
+      props.onSelectionChange(initialValue);
+    }
+  }, []);
```
Comment (sprrwhwk) [inline carbon/components/OptionsSwitcher/index.tsx:42]: We have similar code in `PageWithTabs/index.tsx` maybe we can extract that logic into a library function?

**Review by sprrwhwk — DISMISSED** [review]

File: carbon/components/OptionsSwitcher/index.tsx (hunk 1 above)
Comment (biwano) [reply carbon/components/OptionsSwitcher/index.tsx:31]: And you showed it to me already but i am a slow learner :)

File: carbon/components/OptionsSwitcher/index.tsx (hunk 2 above)
Comment (biwano) [reply carbon/components/OptionsSwitcher/index.tsx]: done

**Comment by jabby09** [comment]
@biwano is this PR updated and ready to test?
i'm still getting a lot of inconsistent results

**Comment by biwano** [comment]
> @biwano is this PR updated and ready to test? i'm still getting a lot of inconsistent results

Hi @jabby09, what do you mean by inconsistent results?

**Pull request merged**